src/auth/oauth2client.ts

```diff
@@ -500,21 +503,26 @@ export class OAuth2Client extends AuthClient {
                 null, response);
             return;
           }
-          const cacheControl = response.headers['cache-control'];
+
+          let cacheControl = response ? response.headers['cache-control'] : '';
+          if (cacheControl instanceof Array) {
+            cacheControl = cacheControl.length > 0 ? cacheControl[0] : '';
```

Comment:
I'm not too familiar with the `cache-control` control header. Under what cases can it be an array, and why is it okay to pick the first one? Does this need a code comment?

Reply:
I switched this up because the NetIncomingHeader type in the http says it's string|string[].  Realistically, this was working before so I just returned it to original form and did a cast.